project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestSingleFileProject.java

```diff
@@ -74,6 +74,21 @@ public void testLoadSingleFile() {
 
     }
 
+    @Test (description = "tests if the target directory for single files is resolved properly")
+    public void testSingleFileTargetDirectory() {
+        Path projectPath = RESOURCE_DIRECTORY.resolve("single_file").resolve("main.bal");
+        SingleFileProject project = null;
+        try {
+            project = TestUtils.loadSingleFileProject(projectPath);
+        } catch (Exception e) {
+            Assert.fail(e.getMessage());
+        }
+
+        Path targetDirPath = project.targetDir();
+        Assert.assertNotNull(targetDirPath);
+        Assert.assertTrue(targetDirPath.toFile().exists());
```

Comment:
Lets add an assertion to make sure its a temp location

Reply:
Added extra assertion : `Assert.assertEquals(Path.of(System.getProperty("java.io.tmpdir")), targetDirPath.getParent());`

Reply:
Changed to us `Paths.get()`